full-service/src/db/txo.rs

```diff
@@ -1847,11 +1848,156 @@ mod tests {
         );
     }
 
+    #[test_with_logger]
+    fn test_max_spendable_mob(logger: Logger) {
+        let mut rng: StdRng = SeedableRng::from_seed([20u8; 32]);
+
+        let db_test_context = WalletDbTestContext::default();
+        let wallet_db = db_test_context.get_db_instance(logger);
+
+        let root_id = RootIdentity::from_random(&mut rng);
+        let account_key = AccountKey::from(&root_id);
+        let (account_id, _address) = Account::create_from_root_entropy(
+            &root_id.root_entropy,
+            Some(0),
+            None,
+            None,
+            "",
+            None,
+            None,
+            None,
+            &wallet_db.get_conn().unwrap(),
+        )
+        .unwrap();
+
+        // Seed Txos
+        let mut src_txos = Vec::new();
+        // total MOB available == 210
+        // max spendable == 200 (20 + 19 + ... + 5), 16 txos total
+        for i in 1..=20 {
+            let (_txo_id, txo, _key_image) =
+                create_test_received_txo(&account_key, i, i as u64, i, &mut rng, &wallet_db);
+            src_txos.push(txo);
+        }
+
+        // testing over the total in account, should fail
+        let result = Txo::select_unspent_txos_for_value(
+            &account_id.to_string(),
+            500 as u64,
+            None,
+            &wallet_db.get_conn().unwrap(),
+        );
+        assert!(result.is_err());
+
+        // testing over the max spendable but under the total in account, should fail
+        let result = Txo::select_unspent_txos_for_value(
+            &account_id.to_string(),
+            201 as u64,
+            None,
```

Comment:
This doesn't match the comment above (max spendable is None here)

Reply:
Resolved over slack, this test checks for max spendable by the account and not by filtering for specific utxos that are under a certain value.

Reply:
update should have clarified this, but max spendable can have 2 meanings. In the context of the select_unspent_txos_for_value function, max_spendable_value is referring to the largest value an individual txo can have to be considered valid for the sake of selecting it for a transaction. In the context of the test, max_spendable_in_account is referring to "what is the maximum spendable amount of MOB that I can have be part of a single transaction"